application/src/main/java/org/opentripplanner/graph_builder/module/osm/OsmDatabase.java

```diff
@@ -102,8 +104,18 @@ public class OsmDatabase {
   /* Set of all node IDs of kept areas. Needed to mark which nodes to keep in stage 3. */
   private final TLongSet areaNodeIds = new TLongHashSet();
 
-  /* Track which vertical level each OSM way belongs to, for building elevators etc. */
-  private final Map<OsmEntity, OsmLevel> wayLevels = new HashMap<>();
+  /* Track which vertical level each OSM way belongs to, e.g. for building elevators.
+   * For backwards compatibility reasons each way has a mapping to a level.
+   * If the level can not be determined, the default level is used.
+   */
+  private final Map<OsmEntity, OsmLevel> waySingleLevelMap = new HashMap<>();
```

Comment:
Have you noticed a large increase in memory for these?

Reply:
Haven't checked, I will look into memory consumption if there is a need for it